Zend/zend_inheritance.c

```diff
@@ -3527,18 +3528,26 @@ ZEND_API zend_class_entry *zend_do_link_class(zend_class_entry *ce, zend_string
 		}
 	}
 
+	size_t num_implementable_interfaces = 0;
 	if (ce->num_interfaces) {
 		for (i = 0; i < ce->num_interfaces; i++) {
 			zend_class_entry *iface = zend_fetch_class_by_name(
 				ce->interface_names[i].name, ce->interface_names[i].lc_name,
 				ZEND_FETCH_CLASS_INTERFACE |
-				ZEND_FETCH_CLASS_ALLOW_NEARLY_LINKED | ZEND_FETCH_CLASS_EXCEPTION);
+				ZEND_FETCH_CLASS_ALLOW_NEARLY_LINKED | ZEND_FETCH_CLASS_EXCEPTION |
+				(ce->interface_names[i].is_optional ? ZEND_FETCH_CLASS_SILENT : 0));
+
+			// Optional interfaces are skipped if they don't exist.
+			if (!iface && ce->interface_names[i].is_optional) {
+				continue;
+			}
+
 			if (!iface) {
 				check_unrecoverable_load_failure(ce);
 				free_alloca(traits_and_interfaces, use_heap);
 				return NULL;
 			}
-			traits_and_interfaces[ce->num_traits + i] = iface;
+			traits_and_interfaces[ce->num_traits + num_implementable_interfaces++] = iface;
```

Comment:
As mentioned previously, I think we should at least set the remaining slots in `traits_and_interfaces` to `NULL` if they are unused. This will prevent use-of-uninitialized-memory issues in `zend_accel_inheritance_cache_find()` and similar places.